serializable.ts

```diff
@@ -179,42 +179,34 @@ function fromJSON<T>(
   const serializablePropertyMap = SERIALIZABLE_CLASS_MAP.get(
     context?.constructor?.prototype,
   );
+
   if (!serializablePropertyMap) {
     throw new Error(
       `${ERROR_MISSING_PROPERTIES_MAP}: ${context?.constructor
         ?.prototype}`,
     );
   }
 
-  const _json = typeof json === "string" ? json : JSON.stringify(json);
+  const _json = typeof json === "string" ? JSON.parse(json) : json;
+  const accumulator: Partial<T> = {};
 
-  return Object.assign(
-    context,
-    JSON.parse(
-      _json,
-      /** Processes the value through the provided or default `fromJSONStrategy` */
-      function revive(key: string, value: JSONValue): unknown {
-        // After the last iteration of the fromJSONStrategy a function
-        // will be called one more time with a empty string key
-        if (key === "") {
-          return value;
-        }
+  for (const [key, value] of Object.entries(_json)) {
+    const {
+      propertyKey,
+      fromJSONStrategy = fromJSONDefault,
+    } = serializablePropertyMap.getBySerializedKey(key) || {};
 
-        const {
-          propertyKey,
-          fromJSONStrategy = fromJSONDefault,
-        } = serializablePropertyMap.getBySerializedKey(key) || {};
+    if (!propertyKey) {
```

Comment:
If we have @SerializeProperty() with no string value (as opposed to @SerializeProperty("something")), will this still be OK (will propertyKey still be defined)?

Reply:
yep! because the key is the property as string if not set